### DISCUSSION TITLE
Help recovering my project

### DISCUSSION THREAD

**Discussion by barodapride** [git-lfs/git-lfs#5659] [Q&A] [answered]
Hi I'm wondering if someone can help me with this. 

I have a project that I was working on (a Unity game dev project), and after I installed git lfs, put in the .gitattributes file (with extensions typical for a Unity project) and ran git lfs migrate import --everything it appears many files are now deleted from my project. Is there a way to recover? I'm not sure what happened. Any ideas on what I can do? I have a git repo locally. When I tried to upload to my github repo I got an error saying one of the files was too big so that's when I decided to install git lfs. However, now after running the git lfs commands and (attempting to) migrate the old commits to git lfs it seems my project is broken and I still cannot commit or upload. I don't know how it's possible some of my files are simply missing. It seems like the bigger ones that got taken by git lfs. Any help is appreciated.

**Comment by bk2204** [comment]
What specific output are you seeing that makes you think that the files are deleted?  Can you also tell us what behaviour you were expecting?

**Reply by barodapride** [reply]
Yes sorry I should have been more clear. I can see that many (if not all) of my files contents got replaced with some git-lfs info like so:

`
version https://git-lfs.github.com/spec/v1
oid sha256:ca3cb4c6d9e96c9e288c67e7f0a0537d2bba09bde43bc335c6b5b04b93bde1a1
size 1635
`

Even small text files - C# source code files got replaced with text like that.
So I guess it tried to convert all files into git lfs files? I'm not 100% clear on what happens to lfs files locally are they still stored somewhere on my computer? I thought they should remain normal locally and only when pushing do they get intercepted. I already ran a git lfs uninstall locally in that directory so maybe the original contents got deleted?

**Reply by bk2204** [reply, marked as answer]
You should be able to do a `git lfs checkout` to convert those into regular files.  However, first, you need to run `git lfs install`, because `git lfs checkout` won't work if you haven't or if you've run `git lfs uninstall`, and it won't push the large files to the remote in that case, either.

The way Git LFS works is that it converts the files you tell it to into pointer files like you see there, and the large files are stored outside of the repository and pushed before the regular Git objects.

**Reply by barodapride** [reply]
I was able to run `git lfs` checkout and it worked! So I guess even though I ran `git lfs uninstall` the files were still stored somewhere locally? Or maybe they were stored remotely in some server? Although I'm not sure how that would work because I hadn't set up a remote git repository yet. In any case, I'll take my time to properly set it up this time and make a backup just in case anything goes wrong again. Thanks!

**Reply by bk2204** [reply]
LFS files that you haven't pushed are stored locally in the `.git` directory.  We don't delete them unless you run `git lfs prune`.  Once you've pushed your code to the server, they can be safely pruned, and then they can be downloaded on demand.

**Reply by walter-322** [reply]
And what if .git folder got deleted? My friend has this Issue
Like everything gone

**Reply by chrisd8088** [reply]
> And what if .git folder got deleted?

I'm not sure what you could do in this case.  If you've (a) not pushed your data to a remote, (b) not made any local backups, and (c) deleted all local copies, then you're probably not going to be able to restore the data, at least not without using OS-level or third-party tools to recover the files.  If your friend is genuinely in this kind of situation and the data is truly important, have them stop using their drive and look into using file-recovery tools appropriate to their operating system.
